wiki_service/mongoAPI.py

```diff
@@ -0,0 +1,78 @@
+from bson import ObjectId
+from pymongo import MongoClient
+from urllib.parse import quote_plus
+from os import getenv
+from http import HTTPStatus
+import logging as log
+
+
+class MongoAPI:
+    def __init__(self, ):
+        # sets up database client
+        user = getenv('MONGODB_USER', "admin")
+        password = getenv('MONGODB_PASSWORD', "test123")
+        databaseHost = getenv('MONGODB_HOST', "mymongo_wiki")
+        uri = "mongodb://%s:%s@%s" % (quote_plus(user), quote_plus(password), databaseHost)
+        # THIS IS JUST FOR TESTING PURPOSES
+        self.client = MongoClient(uri)
+        database = "mindtasticWiki"
+        collection = "articles"
+        db = self.client[database]
+        # creates collection in wiki database
+        if collection not in db.list_collection_names():
+            db.create_collection(collection)
+        self.collection = db[collection]
+
+    def readAll(self):
+        log.info('Reading All Articles')
+        documents = self.collection.find()
+        articles = []
+        for article in documents:
+            print(article)
+            articles.append({
+                "id": str(article["_id"]),
+                "title": article["title"],
+                "content": article["content"]
+            })
+        output = {"articles": articles}
+        return output
+
+    def readSingle(self, articleID):
+        log.info('Reading specific article')
+        document = self.collection.find_one(ObjectId(articleID))
+        document["_id"] = str(document["_id"])
+        document["id"] = document.pop("_id")
+        return document
+
+    def write(self, data):
+        # this method is supposed to replace all existing
+        # entries with the ones coming from the admin service
+
+        # --> admins can just send one json object to this endpoint, which
+        # is updated from time to time
+
+        # 1. Delete all existing entries in collection
+        self.collection.delete_many({})
```

Comment:
I have a couple of questions regarding the new write method. If I understand it correctly, the only way to add new articles is by overwriting all existing articles with new ones. However, the first check here returns if there are documents in the collection?